### ISSUE TITLE
Writing in small cursive ends up in having the word replaced by a straight line

### ISSUE THREAD

**Issue by c-beuschel** [saber-notes/saber#1186]
### Before submitting an issue, please check the following

- [X] I have searched for similar issues (both open and closed) and cannot find a duplicate
- [X] I agree to follow the [Code of Conduct](https://github.com/saber-notes/saber/blob/main/CODE_OF_CONDUCT.md)

### Describe the bug

I did a long writing session on my Google Pixel Tablet with GrapheneOS on Saber v0.21.2 F-Droid (21020) with an USI 2.0 Pen (Line thickness "2", Line height 60).

I write mostly in traditional and sloppy German cursive and often without lifting the pen for a whole word.

Everytime I wrote a word that was about 5mm in height and 15mm or more in length with a right ponting starting and end stroke, my word was converted to a straight line.

![2024-03-21-15-47-16-406](https://github.com/saber-notes/saber/assets/69590199/cf6b3404-08b3-405c-abba-393dc8f40aea)

### To reproduce

![2024-03-21-15-47-16-406](https://github.com/saber-notes/saber/assets/69590199/de41811e-4a5f-4ba0-850a-c83896c48cb3)

The picure shows two examples of writing. Both done on paper and on the tablet. The first one is fine, because it starts with a downward stroke an ends with a stroke to the right. The second one starts and ends with the pen moving in the same direction and gets converted to a staight line.

### Expected behavior

Assistans with straight lines is nice and all, but there is much change in direction in the given example. That schould prevent the algorithm from "helping with straight lines".

### Saber version

v0.21.2 F-Droid (21020)

### Device

- Device: Google Pixel Tablet
- OS: GrapheneOS (Android)

### Anything else?

Thank you very much for developing Saber as an Free Open Source Applikation.

**Comment by QubaB** [comment]
You should turn off Auto staighten lines in settings
![image](https://github.com/saber-notes/saber/assets/74395874/447830da-6277-4331-97be-cb05d8fd0831)

I do not use it too.

**Comment by c-beuschel** [comment]
That might help with my writing, thank you.

It does not resolve the issue. There schouldnt be a straight line, when drawn line changes the angle of its vector by more than 90° relative to the start and end of the line at any point.

**Comment by QubaB** [comment]
> There shouldnt be a straight line, when drawn line changes the angle of its vector by more than 90° relative to the start and end of the line at any point.

Yes, it is probably better way to compare angles of segments of line. But if the segment will be very short (caused by uncertainty of position), it can be problematic to distinguish if it is straight line or handwriting.

**Comment by QubaB** [comment]
I changed algortihm of detection in #1187 to suggestion of @c-beuschel
